vector-im/element-android
# Adds up navigation in spaces

## Type of change


## Content

Being in a space or subspace now allows to click an up button on the toolbar or the device back button to navigate up through the spaces

## Motivation and context

Closes https://github.com/vector-im/element-android/issues/3958

I added a new vector icon for the back button instead of using the default inbuilt toolbar one as given we're using a custom toolbar and it very dynamically changes based on space state it's easier to manage it this way. On the topic of using the existing icon, we do have a back icon but the arrow is much longer and stands out as not being the android native icon.

## Screenshots / GIFs

https://user-images.githubusercontent.com/20701752/169823500-27863d55-aef1-45e3-a06d-c1874d6fc122.mp4

## Tests

1. Navigate into a space
2. Click the newly appearing up button on the toolbar
3. Do the same but instead of the up button use the device's back button
4. Also try doing this in subspaces. This should work across any number of layers

## Tested devices

- OS version(s): Android 12

## Checklist
